src/dune_rules/stanza_common.mli

```diff
@@ -1,19 +1,31 @@
 open Import
+open Dune_lang.Decoder
 
 module Pkg : sig
-  val decode : Package.t Dune_lang.Decoder.t
+  val decode : Package.t t
 
   val resolve :
     Dune_project.t -> Package.Name.t -> (Package.t, User_message.t) Result.t
 
-  val field : stanza:string -> Package.t Dune_lang.Decoder.fields_parser
+  val field : stanza:string -> Package.t fields_parser
 
-  val field_opt :
-       ?check:unit Dune_lang.Decoder.t
-    -> unit
-    -> Package.t option Dune_lang.Decoder.fields_parser
+  val field_opt : ?check:unit t -> unit -> Package.t option fields_parser
 
   val default_exn : loc:Loc.t -> Dune_project.t -> string -> Package.t
 end
 
-val modules_field : string -> Ordered_set_lang.t Dune_lang.Decoder.fields_parser
+val modules_field : string -> Ordered_set_lang.t fields_parser
+
+(** [preprocess] and [preprocessor_deps] fields *)
+val preprocess_fields :
+  ( Preprocess.Without_instrumentation.t Preprocess.Per_module.t
+    * Dep_conf.t list
+  , fields )
+  parser
```

Comment:
We have the alias `fields_parser` to simplify this.

Reply:
https://github.com/ocaml/dune/pull/6349/commits/e8240d0b8838ffa4dab4703821184313932e4b4f